[Term Entry] Python Pillow - Image: .merge() (#6355)
Codecademy/docs: content/pillow/concepts/image/terms/merge/merge.md

---
Title: '.merge()'

> Description: 'Set of single band images into a new multiband image'

Reviewer: As per the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md), `Description`: for a coding term should begin with a present-tense verb, i.e., "Returns", "Calculates", "Outputs", etc.

```suggestion
Description: 'Merges set of single-band images into a new multi-band image.'
```

Subjects:
  - 'Computer science'
  - 'Data Science'
  - 'Data Visualization'
Tags:
  - 'Computer Vision'
  - 'Images'
  - 'Pillow'
  - 'Values'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
  - 'paths/data-science'
  - 'paths/data-science-foundations'
---

> The **`.merge()`** is used to set of single band images into a new multiband image. We can create a new image by merging specific bands when we working with multispectral or multichannel images such as RGB or CMYK images.

Reviewer: This can written better as -

```suggestion
The **`.merge()`** function in Pillow combines multiple single-band images into a new multi-band image. This is particularly useful when working with multispectral or multichannel images, such as RGB or CMYK images, where individual channels can be processed separately before merging them into a final image.
```

## Syntax

```pseudo
Image.merge(mode, bands)
```

> - `mode` : The parameter is the mode of the new multiband image.

Reviewer: ```suggestion
- `mode`: The mode of the new multiband image (e.g., "RGB", "CMYK", etc.).
```

> - `bands` :This parameter is a tuple of the individual image bands that is to be merged. The individual images should be single-channel image or a grayscale image.

Reviewer: ```suggestion
- `bands`: A tuple containing the individual image bands to be merged. Each band should be a single-channel (grayscale) image.
```

## Example

> The code below splits the image into single bands and merge them.

Reviewer: ```suggestion
The code below splits an image into its individual bands and then merges them back in a different order.
```

> ![Boston Skyline](../../../../../../media/Boston.jpg)

Reviewer: Images should be linked to the URL: ["https://raw.githubusercontent.com/Codecademy/docs/main/media/"](https://raw.githubusercontent.com/Codecademy/docs/main/media/) followed by the image name.

```suggestion
![Boston Skyline](https://raw.githubusercontent.com/Codecademy/docs/main/media/Boston.jpg)
```

```py

> from PIL import Image
> image1= Image.open('media/Boston.jpg')
> image1.load()
> im = image1.resize((400,400))
> 
> r,g,b = im.split()
> new_image= Image.merge('RGB', (b,r,g))
> new_image.show()

Reviewer: Add comments and spaces for better understanding -

```suggestion
from PIL import Image

# Open the image
image = Image.open("media/Boston.jpg")
image = image.resize((400, 400))

# Split the image into its RGB channels
r, g, b = image.split()

# Merge the bands in a different order (swapping red and blue)
new_image = Image.merge("RGB", (b, r, g))
new_image.show()
```

```

> **The output image**

Reviewer: ```suggestion
The output will be:
```

> ![Merged Boston Skyline](../../../../../../media/merged_boston.png)

Reviewer: Images should be linked to the URL: ["https://raw.githubusercontent.com/Codecademy/docs/main/media/"](https://raw.githubusercontent.com/Codecademy/docs/main/media/) followed by the image name.

File names should use dashes (`-`) to separate words rather than spaces or underscores. Please update file name as well.

```suggestion
![Merged Boston Skyline](https://raw.githubusercontent.com/Codecademy/docs/main/media/merged-boston.png)
```
